samples/server/petstore/php-slim4/lib/Mock/OpenApiDataMockerInterface.php

```diff
@@ -0,0 +1,175 @@
+
+    /** @var string DATA_FORMAT_EMAIL */
+    public const DATA_FORMAT_EMAIL = 'email';
+
+    /** @var string DATA_FORMAT_UUID */
+    public const DATA_FORMAT_UUID = 'uuid';
+
+    /**
+     * Mocks OpenApi Data.
+     * @see https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.1.md#data-types
+     *
+     * @param $dataType   string     OpenApi data type. Use constants from this class
+     * @param $dataFormat string     (optional) OpenApi data format
+     * @param $options    array|null (optional) OpenApi data options
+     *
+     * @throws \InvalidArgumentException when invalid arguments passed
+     *
+     * @return mixed
+     */
+    public function mock(
+        $dataType,
+        $dataFormat = null,
+        $options = []
+    );
+
+    /**
+     * Shortcut to mock integer type
+     * Equivalent to mockData(DATA_TYPE_INTEGER);
+     *
+     * @param string|null $dataFormat       (optional) int32 or int64
+     * @param number|null $minimum          (optional) Default is 0
+     * @param number|null $maximum          (optional) Default is mt_getrandmax()
+     * @param bool|null   $exclusiveMinimum (optional) Default is false
+     * @param bool|null   $exclusiveMaximum (optional) Default is false
+     *
+     * @throws \InvalidArgumentException when $maximum less than $minimum or invalid arguments provided
+     *
+     * @return int
+     */
+    public function mockInteger(
+        $dataFormat = null,
+        $minimum = null,
+        $maximum = null,
+        $exclusiveMinimum = false,
+        $exclusiveMaximum = false
+    );
+
+    /**
+     * Shortcut to mock number type
+     * Equivalent to mockData(DATA_TYPE_NUMBER);
+     *
+     * @param string|null $dataFormat       (optional) float or double
+     * @param number|null $minimum          (optional) Default is 0
+     * @param number|null $maximum          (optional) Default is mt_getrandmax()
+     * @param bool|null   $exclusiveMinimum (optional) Default is false
+     * @param bool|null   $exclusiveMaximum (optional) Default is false
+     *
+     * @throws \InvalidArgumentException when $maximum less than $minimum or invalid arguments provided
+     *
+     * @return float
+     */
+    public function mockNumber(
+        $dataFormat = null,
+        $minimum = null,
+        $maximum = null,
+        $exclusiveMinimum = false,
+        $exclusiveMaximum = false
+    );
+
+    /**
+     * Shortcut to mock string type
+     * Equivalent to mockData(DATA_TYPE_STRING);
+     *
+     * @param string|null $dataFormat (optional) one of byte, binary, date, date-time, password
+     * @param int|null    $minLength  (optional) Default is 0
+     * @param int|null    $maxLength  (optional) Default is 100 chars
+     *
+     * @throws \InvalidArgumentException when invalid arguments passed
+     *
+     * @return string
+     */
+    public function mockString(
+        $dataFormat = null,
+        $minLength = 0,
+        $maxLength = null
+    );
```

Comment:
Should I add `$pattern = null` argument to `mockString()` method? I won't implement it right away, but it helps to overcome breaking changes in future.

Reply:
Sounds like a good idea to me and we can leave a `TODO` next to it.

Besides pattern, here are some other useful properties:
```
maximum
exclusiveMaximum
minimum
exclusiveMinimum
maxLength
minLength
pattern (This string SHOULD be a valid regular expression, according to the ECMA 262 regular expression dialect)
maxItems
minItems
uniqueItems
maxProperties
minProperties
```



Ref: https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.0.md#properties